### DISCUSSION TITLE
Using (Entendendo Conceito)

### DISCUSSION THREAD

**Discussion by cmsouzaac** [balta-io/2801#87] [Perguntas e Respostas] [answered]
Professor pelo que entendi quando criamos uma nova pasta com uma nova class precisamos declarar no código principal a palavra using, para assim a class criada possa puxar todas configurações da pasta principal. Seria esse entendimento?

**Comment by andrebaltieri** [comment, marked as answer]
@cmsoouza bom dia meu querido!!
As pastas são divisão física e os `namespaces` a divisão lógica.

É uma boa prática os namespaces seguirem o mesmo caminho das pastas... então por exemplo, se temos a seguinte estrutura:
```
C:\dev\Blog\Models\Post.cs
```
Seria bacana na classe `Post.cs` ter o namespace `namespace Blog.Models`

Desta forma, sempre que precisar referenciar esta classe em outra classe que esteja fora deste namespace (Ou pasta caso esteja seguindo nossa convenção), precisa do `usign`, como por exemplo:

```csharp
using Blog.Models; // Referencia o namespace dos models

namespace Blog.Repositories;

public class PostRepository 
{
    public List<Post> Posts { get; set; }
}

```
